diplo.lua

```diff
@@ -0,0 +1,128 @@
+@@ -0,0 +1,79 @@
+--[[
+diplo - Quick "Diplomacy"
+Without arguments:
+    Print list of civs the player civ has diplomatic relations to.
+    This list might not match the ingame civ list, but is what matters. Does not include 'no contact' civs, etc.
+    Also shows if relations match between player civ and other civ - a mismatch can potentially happen 'naturally', but is important when trying to make peace proper, as both civs need to be at peace with the other.
+With arguments:
+    diplo CIV_ID RELATION
+    Changes the relation to the civ identified by CIV_ID to the one specified in RELATION, making sure that this is mutual.
+    CIV_ID can be 'all' to change relations with all civs.
+    RELATION can be 'peace'/0 or 'war'/1.
+    Civ list is shown afterwards.
+]]
+
+local args = {...}
+
+-- player civ references:
+local p_civ_id = df.global.plotinfo.civ_id
+local p_civ = df.historical_entity.find(df.global.plotinfo.civ_id)
+
+-- get list of civs:
+function get_civ_list()
+    local civ_list = {}
+    for _, entity in pairs(p_civ.relations.diplomacy) do
+        local cur_civ_id = entity.group_id
+        local cur_civ = df.historical_entity.find(cur_civ_id)
+        if cur_civ.type == 0 then
+            -- if true then
+            rel_str = ""
+            if entity.relation == 0 then
+                rel_str = "0 (Peace)"
+            elseif entity.relation == 1 then
+                rel_str = "1 (War)"
+            end
+            matched = "No"
+            for _, entity2 in pairs(cur_civ.relations.diplomacy) do
+                if entity2.group_id == p_civ_id and entity2.relation == entity.relation then
+                    matched = "Yes"
+                end
+            end
+            table.insert(civ_list, {
+            cur_civ_id,
+            rel_str,
+            matched,
+            dfhack.TranslateName(cur_civ.name, true)
+            })

```

Comment:
it is difficult to remember an arbitrary order of values when you come back and modify this script later. it is much more programmer-friendly to give them names, e.g.:
```lua
table.insert(civ_list, {
    civ_id=cur_civ_id,
    relation=entity.relation,
    matched=matched,
    name=dfhack.TranslateName(cur_civ.name, true),
})